core/controllers/editor.py

```diff
@@ -56,6 +54,25 @@ def _require_valid_version(version_from_payload, exploration_version):
             % (exploration_version, version_from_payload))
 
 
+# Common schemas used in this file.
+SCHEMA_FOR_EXPLORATION_ID = {
+    'type': 'basestring',
+    'validators': [{
+        'id': 'has_length_at_most',
+        'max_value': 12 # Max length of an exploration_id
+    }, {
+        'id': 'is_nonempty'
+    }]
```

Comment:
I have updated the validator for exploration_id -- now it checks with a regex.
Also it will check & pass for the sample id's as mentioned by you.
![image](https://user-images.githubusercontent.com/57531197/128077165-56a51f3a-5fbb-40c8-833a-588633838eed.png)